modules/custom/social_scroll/src/SocialScrollOverride.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\social_scroll;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ConfigFactoryOverrideInterface;
+use Drupal\Core\Config\StorageInterface;
+
+/**
+ * Class SocialScrollOverride.
+ *
+ * @package Drupal\social_scroll
+ */
+class SocialScrollOverride implements ConfigFactoryOverrideInterface {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The SocialScrollManager manager.
+   *
+   * @var \Drupal\social_scroll\SocialScrollManager
+   */
+  protected $socialScrollManager;
+
+  /**
+   * Constructs the configuration override.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\social_scroll\SocialScrollManager $social_infinite_scroll_manager
+   *   The SocialScrollManager manager.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, SocialScrollManager $social_infinite_scroll_manager) {
+    $this->configFactory = $config_factory;
+    $this->socialScrollManager = $social_infinite_scroll_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function loadOverrides($names) {
+    $overrides = [];
+    $enabled_views = $this->socialScrollManager->getEnabledViewIds();
```

Comment:
```suggestion
    $enabled_views = $this->socialScrollManager->getEnabledViewIds();

    $scroll_config = $this->configFactory->get('social_scroll.settings');
    $button_text = $scroll_config->get('button_text');
    $automatically_load_content = $scroll_config->get('automatically_load_content');
```

Reply:
Moved in condition to not execute code on every request